submissions/AnnaStoyano/Friends-app/index.js

```diff
@@ -0,0 +1,169 @@
+            const friendsApp = new FriendsApp(data.results);
+            friendsApp.start();
+        })
+        .catch(() => {
+            document.location.reload();
+        })
+});
+
+class FriendsApp {
+    constructor(users) {
+        this.resultsWrapper = document.querySelector('.result');
+        this.cards = users.map(item => new UserCard(item));
+        this.dipslayCards = this.cards;
+        this.previouseDisplay = this.dipslayCards;
+        this.clickedCount = 0;
+        this.filterByName();
+        this.filterByAge();
+        this.filterByGender();
+        this.filterByCity();
+        this.filterByEmail();
+        this.reset();
+    }
+
+    start() {
+        this.render();
+    }
+
+    render() {
+        this.resultsWrapper.innerHTML = '';
+        this.dipslayCards.forEach(({card}) => this.resultsWrapper.insertAdjacentElement('afterbegin', card));
+    }
+
+    filterByName() {
+        const nameWrapper = document.querySelector('.menu_item-name');
+        nameWrapper.addEventListener('click', ({target}) => {
+            if (target.classList.contains('name-alphabetically')) {
+                this.dipslayCards = this.dipslayCards.sort((a, b) => a.name > b.name ? -1 : 1);
+            }
+            if (target.classList.contains('name-nonalphabetically')) {
+                this.dipslayCards = this.dipslayCards.sort((a, b) => a.name < b.name ? -1 : 1);
+            }
+            this.render();
+        });
+    }
+
+    filterByAge() {
+        const ageButton = document.querySelector('.age_btn');
+        ageButton.addEventListener('click', () => {
+            if (ageButton.classList.contains('clicked')) {
+                this.dipslayCards = this.dipslayCards.sort((a, b) => a.age < b.age ? -1 : 1);
+                ageButton.classList.remove('clicked');
+            } else {
+                this.dipslayCards = this.dipslayCards.sort((a, b) => a.age > b.age ? -1 : 1);
+                ageButton.classList.add('clicked');
+            }
+            this.render();
+        });
+    }
+
+    filterByGender() {
+        const genderWrapper = document.querySelector('.menu_item-gender');
+        genderWrapper.addEventListener('click', ({target}) => {
+            if (this.clickedCount == 0) {
+                this.previouseDisplay = this.dipslayCards;
+            }
+            if (target.value != 'all' && target.value) {
+                this.dipslayCards = this.previouseDisplay.filter(card => card.user.gender == target.value);
+                this.clickedCount++;
+            } else {
+                this.dipslayCards = this.previouseDisplay;
+            }
+            this.render();
+        });
+    }
+
+    filterByCity() {
+        const cityInput = document.querySelector('.text-input--city');
+        const cityWrapper = document.querySelector('.menu_item-city');
+        cityWrapper.addEventListener('click', () => {
+            filterByInputText.bind(this)(cityInput, cityWrapper);
```

Comment:
Why you are using `bind` here?

Reply:
Because `filterByInputText` function is outside the class I bind app(this) to  `this` inside function

Reply:
I see. Missed this part. Thank you for an explanation.

But in general, this is antipattern because you are providing unclear dependency - if you will look in your `filterByInputText `, you will don't understand what is `this.dipslayCards`. If you want have something in your function, which isn't available in the scope of this function, you should pass it as argument.

Also, your function will know about structure of your class - this is violation of encapsulation of your class.

I will suggest you add `filterByInputText ` as a method.